src/Metadata/Property/Factory/DefaultPropertyMetadataFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property\Factory;
+
+use ApiPlatform\Core\Exception\PropertyNotFoundException;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+
+class DefaultPropertyMetadataFactory implements PropertyMetadataFactoryInterface
+{
+    private $decorated;
+
+    public function __construct(PropertyMetadataFactoryInterface $decorated = null)
+    {
+        $this->decorated = $decorated;
+    }
+
+    public function create(string $resourceClass, string $property, array $options = []): PropertyMetadata
+    {
+        if (null === $this->decorated) {
+            $propertyMetadata = new PropertyMetadata();
+        } else {
+            try {
+                $propertyMetadata = $this->decorated->create($resourceClass, $property, $options);
+            } catch (PropertyNotFoundException $propertyNotFoundException) {
+                $propertyMetadata = new PropertyMetadata();
+            }
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($resourceClass);
+        } catch (\ReflectionException $reflectionException) {
+            return $propertyMetadata;
+        }
+
+        $defaultProperties = $reflectionClass->getDefaultProperties();
+
+        if (array_key_exists($property, $defaultProperties) && null !== ($defaultProperty = $defaultProperties[$property])) {
+            $attributes = $propertyMetadata->getAttributes() ?? [];
+            $attributes['swagger_context']['default'] = $defaultProperty;
```

Comment:
The default value could be useful in other formats (Hydra for instance). I would introduce a new, dedicated, attribute for this. The other benefit of doing this is that it would be possible to set a default value that is different than the one of the property by using the `@ApiProperty(default="foo bar")` annotation.